app/src/main/java/org/wikipedia/analytics/eventplatform/EditAttemptStepEvent.kt

```diff
@@ -42,10 +44,13 @@ class EditAttemptStepEvent(private val event: EditAttemptStepInteractionEvent) :
         private fun submitEditAttemptEvent(action: String, editorInterface: String, pageTitle: PageTitle) {
             EventPlatformClient.submit(EditAttemptStepEvent(EditAttemptStepInteractionEvent(action,
                 WikipediaApp.instance.appInstallID, "", editorInterface,
-                INTEGRATION_ID, "", WikipediaApp.instance.getString(R.string.device_type).lowercase(), 0,
-                    if (AccountUtil.isLoggedIn) AccountUtil.getUserIdForLanguage(pageTitle.wikiSite.languageCode) else 0,
+                INTEGRATION_ID, "", WikipediaApp.instance.getString(R.string.device_type).lowercase(), 0, getUserIdForWikiSite(pageTitle.wikiSite),
                 1, pageTitle.prefixedText, pageTitle.namespace().code())))
         }
+
+        private fun getUserIdForWikiSite(wikiSite: WikiSite): Int {
+            return if (AccountUtil.isLoggedIn) SharedPreferenceCookieManager.instance.getCookieByName("UserID", wikiSite.authority(), false)?.toIntOrNull() ?: 0 else 0
```

Comment:
There is a more efficient way: extract the UserID directly from the cookie set by the current wiki subdomain. That's it -- none of the previous userId management code in AccountUtils is necessary anymore.

Reply:
This is great!